iceoryx_utils/include/iceoryx_utils/cxx/helplets.hpp

```diff
@@ -190,6 +206,43 @@ static constexpr uint64_t strlen2(char const (&/*notInterested*/)[SizeValue])
     return SizeValue - 1;
 }
 
+/// @brief struct to find the best fitting unsigned integer type
+template <bool GreaterUint8, bool GreaterUint16, bool GreaterUint32>
+struct bestFittingTypeImpl
+{
+    using Type_t = uint64_t;
+};
+
+template <>
+struct bestFittingTypeImpl<false, false, false>
+{
+    using Type_t = uint8_t;
+};
+
+template <>
+struct bestFittingTypeImpl<true, false, false>
+{
+    using Type_t = uint16_t;
+};
+
+template <>
+struct bestFittingTypeImpl<true, true, false>
+{
+    using Type_t = uint32_t;
+};
+
+/// @brief get the best fitting unsigned integer type for a given value at compile time
+template <uint64_t Value>
+struct bestFittingType
+{
+    using Type_t = typename bestFittingTypeImpl<(Value > std::numeric_limits<uint8_t>::max()),
+                                                (Value > std::numeric_limits<uint16_t>::max()),
+                                                (Value > std::numeric_limits<uint16_t>::max())>::Type_t;
```

Comment:
c&p error. The last one should probably be `uint32_t`. Could probably be caught by tests ;)